nats.go

```diff
@@ -2592,6 +2745,19 @@ func (nc *Conn) clearPendingFlushCalls() {
 	nc.pongs = nil
 }
 
+// This will clear any pending Request calls.
+// Lock is assumed to be held by the caller.
+func (nc *Conn) clearPendingRequestCalls() {
+	if nc.respMap == nil {
+		return
+	}
+	for _, ch := range nc.respMap {
+		if ch != nil {
+			close(ch)
+		}
```

Comment:
That's because the channel was closed, but left in the map, then respHandler tried to close again.